controllers/users.ts

```diff
@@ -36,4 +37,44 @@ const getSelfData = (
   }
 };
 
-export { getSelfData };
+const patchSelfData = async (
+  req: Request,
+  res: Response
+): Promise<Response<any, Record<string, any>> | Express.BoomError<null>> => {
+  try {
+    const { userData } = req;
+    const userId = userData?.id;
+    const onboardingUserData: any = userData?.onboarding;
+    const { firstname, lastname, bio, username, onboarding } = req.body;
+    const { isOnboarded } = onboarding ?? {};
+
+    // Allowed fields to edit from patch call
+    const data = {
+      firstname,
+      lastname,
+      bio,
+      username,
+      onboarding: {
+        isOnboarded:
```

Comment:
Will this manage merging of the `onboarding` JSON object?
The current implementation adds only the `isOnboarded` field. Can we set it to dynamically update the `onboarding` object, so that we dont need to update the code to add any other onboarding key?
https://github.com/prisma/prisma/discussions/3070